code/test/ressources/mock.ts

```diff
@@ -96,3 +94,21 @@ export const eventData: APIGatewayProxyEvent = {
     body: null,
     isBase64Encoded: false
 };
+
+interface IMockFactory {
+    action: string,
+    code?: string,
+    fight?: string,
+    encounterID?: string
+}
+
+export function MockFactory({code, action, fight, encounterID}: IMockFactory) {
+    if(action === 'boss'){
+        eventData.queryStringParameters = { code: code, action: action};
```

Comment:
This is bad practice. Editing the base event inplace can lead to unexpected behaviour (i.e. in one test you change the http method or smthg like that; in the next test the old configuration would stay, which is not aligned to the [principles of unit testing](https://esj.com/Articles/2012/09/24/Better-Unit-Testing.aspx?m=1&Page=2) see #2

Rather than that, you could initialize the object anew in line 106 `const mockEvent = {...eventData}` and set its properties afterwards.